spec/rspec/core/memoized_helpers_spec.rb

```diff
@@ -363,6 +363,97 @@ def not_ok?; false; end
         expect(subject).to eq(3)
       end
     end
+
+    describe 'threadsafety', :threadsafe => true do
+      # Force call to eq to happen in threaded env to prevent "circular require" warnings
+      # Not actually sure why calling eq in two threads at once causes this (it's a sporadic failure):
+      #
+      # /Users/josh/.gem/ruby/2.2.0/bundler/gems/rspec-expectations-c7f0054e40a1/lib/rspec/matchers.rb:527: warning: loading in progress, circular require considered harmful - /Users/josh/.gem/ruby/2.2.0/bundler/gems/rspec-expectations-c7f0054e40a1/lib/rspec/matchers/built_in/eq.rb
+      # \tfrom /Users/josh/ref/ruby/rspec-core/spec/support/thread_order_support.rb:36:in  `block (2 levels) in pass_to'
+      # \tfrom /Users/josh/ref/ruby/rspec-core/spec/support/thread_order_support.rb:36:in  `call'
+      # \tfrom /Users/josh/ref/ruby/rspec-core/spec/rspec/core/memoized_helpers_spec.rb:402:in  `block (6 levels) in <module:Core>'
+      # \tfrom /Users/josh/.gem/ruby/2.2.0/bundler/gems/rspec-expectations-c7f0054e40a1/lib/rspec/matchers.rb:527:in  `eq'
+      before(:all) { eq 1 }
+
+      class RaiseOnFailuresReporter < RSpec::Core::NullReporter
+        def self.example_failed(example)
+          raise example.exception
+        end
+      end
+
+      def describe_successfully(&describe_body)
+        example_group    = RSpec.describe(&describe_body)
+        ran_successfully = example_group.run RaiseOnFailuresReporter
+        expect(ran_successfully).to eq true
+      end
```

Comment:
> Actually, is that `RSpec.describe(&describe_body)` right?

Yep.  We use this pattern all over the place in rspec-core's specs.  It doesn't add it to the main test suite do to our sandboxing:

https://github.com/rspec/rspec-core/blob/95609709de8c5df85867feda0b9573b5e53a14bd/lib/rspec/core/sandbox.rb
https://github.com/rspec/rspec-core/blob/95609709de8c5df85867feda0b9573b5e53a14bd/spec/support/sandboxing.rb

That ensures that every example runs with a fresh copy of `RSpec.world` such that `RSpec.describe` registers the example group on the temporary, one-off world.